src/meta/src/stream/stream_graph/fragment.rs

```diff
@@ -682,6 +696,74 @@ impl StreamFragmentGraph {
             )
         })
     }
+
+    /// Collect the mapping from table / `source_id` -> `fragment_id`
+    pub fn collect_backfill_mapping(&self) -> HashMap<u32, Vec<FragmentId>> {
+        let mut mapping = HashMap::new();
+        for (fragment_id, fragment) in &self.fragments {
+            let fragment_id = fragment_id.as_global_id();
+            let fragment_mask = fragment.fragment_type_mask;
+            let candidates = [
+                FragmentTypeFlag::CrossDbSnapshotBackfillStreamScan,
+                FragmentTypeFlag::SnapshotBackfillStreamScan,
+                FragmentTypeFlag::StreamScan,
+                FragmentTypeFlag::SourceScan,
```

Comment:
Though `SourceScan` is considered here, we haven't supported backfill order control in source backfill executor yet. Is this expected?

Reply:
If not, shall we return with error if the user create mv ddl specifies backfill order control on any source?

Reply:
We currently ban it in binding phase. I have refined the error message subsequently and method names https://github.com/risingwavelabs/risingwave/pull/20967/commits/9810a40ed2806ec238f1913f8308e0c36dd50434

Reply:
Also removed all related source scan code: https://github.com/risingwavelabs/risingwave/pull/20967/commits/7c15dd44bf3d2e520f65e73002fe39be2b2b2c26